lib/rucio/core/request.py

```diff
@@ -1390,6 +1396,20 @@ def observe(
                 if state == RequestState.DONE:
                     record.files_done += 1
                     record.bytes_done += file_size
+
+                    if submitted_at is not None and started_at is not None:
+                        wait_time = (started_at - submitted_at).total_seconds()
+
+                        # avg_wait_time averages across a tumbling window, so reset avg every few hours
+                        hours_window = 6
+                        if self.avg_wait_time_start_timestamp < (now - datetime.timedelta(hours=hours_window)):
```

Comment:
We decided to use tumbling window for calculating our running average. While sliding window may make more sense, it requires storing all the transfers in the past X hours, and additional computation is required to remove old transfers and adding in new transfers. Therefore, tumbling window is the most feasible.